ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -28,7 +28,7 @@ def reset_local_buffers(self) -> None:
     def append_to_update_buffer(
         self,
         update_buffer: AgentBuffer,
-        agent_id: str,
+        agent_id: int,
```

Comment:
@ervteng Is this correct now? mypy was complaining about demo_loader calling this with `0` instead of a string.

Reply:
Demo loader is wrong, it uses an int for the update buffer. However in env_manager that int is turned into a string, so the `int` won't work in the general case.

Reply:
Actually it's both, in different parts of the code :P

Reply:
Made it a Union[int, str] for now